src/NimiqClient.php

```diff
@@ -0,0 +1,664 @@
+    public function setPool($pool)
+    {
+        return $this->request('pool', $pool);
+    }
+
+    /**
+     * Returns the connection state to mining pool.
+     *
+     * @return int mining pool connection state (0: connected, 1: connecting, 2: closed)
+     */
+    public function getPoolConnectionState()
+    {
+        return $this->request('poolConnectionState');
+    }
+
+    /**
+     * Returns the confirmed mining pool balance.
+     *
+     * @return float confirmed mining pool balance (in smallest unit)
+     */
+    public function getPoolConfirmedBalance()
+    {
+        return $this->request('poolConfirmedBalance');
+    }
+
+    /**
+     * Returns instructions to mine the next block. This will consider pool instructions when connected to a pool.
+     *
+     * @param string $address      address to use as a miner for this block. this overrides the address provided during startup or from the pool.
+     * @param string $extraDataHex hex-encoded value for the extra data field. this overrides the address provided during startup or from the pool.
+     *
+     * @return array mining work instructions
+     */
+    public function getWork($address = null, $extraDataHex = null)
+    {
+        return $this->request('getWork', $address, $extraDataHex);
+    }
+
+    /**
+     * Returns a template to build the next block for mining. This will consider pool instructions when connected
+     * to a pool.
+     *
+     * @param string $address      address to use as a miner for this block. this overrides the address provided during startup or from the pool.
+     * @param string $extraDataHex hex-encoded value for the extra data field. this overrides the address provided during startup or from the pool.
+     *
+     * @return array mining block template
+     */
+    public function getBlockTemplate($address = null, $extraDataHex = null)
+    {
+        return $this->request('getBlockTemplate', $address, $extraDataHex);
+    }
+
+    /**
+     * Submits a block to the node. When the block is valid, the node will forward it to other nodes in the network.
+     *
+     * @param string $blockHex hex-encoded full block (including header, interlink and body). when submitting work from getWork, remember to include the suffix.
+     */
+    public function submitBlock($blockHex)
+    {
+        $this->request('submitBlock', $blockHex);
+    }
+
+    /**
+     * Returns a list of addresses owned by client.
+     *
+     * @return Account[] array of accounts owned by the client
+     */
+    public function getAccounts()
+    {
+        $result = $this->request('accounts');
+
+        return array_map(function ($rawAccount) {
+            switch ($rawAccount['type']) {
+                case AccountType::Vesting:
+                    return new VestingContract($rawAccount);
+                    break;
+
+                case AccountType::Htlc:
+                    return new HashedTimeLockedContract($rawAccount);
+                    break;
+
```

Comment:
Great! However, I belive when using `return`, you don't need to `break` anymore ;)